Library/Homebrew/cli/parser.rb

```diff
@@ -208,11 +208,15 @@ def switch(*names, description: nil, replacement: nil, env: nil, depends_on: nil
         return if global_switch
 
         description = option_description(description, *names, hidden:)
-        process_option(*names, description, type: :switch, hidden:) unless disable
-
+        env, counterpart = env
+        if env && @non_global_processed_options.any?
+          affix = counterpart ? " and `#{counterpart}` is passed." : "."
+          description += " Enabled by default if `$HOMEBREW_#{env.upcase}` is set#{affix}"
+        end
```

Comment:
Is it worth making this flexible so the order doesn't matter? Something like this (not tested):

```suggestion
        if env && @non_global_processed_options.any?
          conditions = env.map do |e|
            if env.is_a? Symbol
              "`$HOMEBREW_#{env.upcase}` is set"
            elsif env.startswith? "--"
            	"`#{counterpart}` is passed"
            end
          end
          description += " Enabled by default if {conditions.to_sentence}."
        end
```

Reply:
Since it currently supports only one instance, my plan was to leave it in this simpler form for now and expand it later if the need arises.